index.js

```diff
@@ -85,10 +85,21 @@ async function pay (plugin, {
 
     const payStream = ilpConn.createStream()
 
-    return Promise.race([
-      payStream.sendTotal(sendAmount).then(() => payStream.end()),
-      new Promise(resolve => payStream.on('end', resolve))
-    ])
+    try {
+      await payStream.sendTotal(sendAmount, { timeout: streamOpts.timeout })
+    } catch (err) {
+      const totalSent = payStream.totalSent
+      try {
+        await ilpConn.end()
+      } catch (err) {
+        logger.debug('Error while ending connection:', err)
+      }
+      throw new PaymentError('Failed to send specified amount', { totalSent })
+    }
+
+    const totalSent = payStream.totalSent
+    await ilpConn.end()
```

Comment:
I'm wondering if this should also be wrapped in a try/catch like above (ditto for `pull` below)

Reply:
I don't think it has to because we only get there when the `sendTotal` and `receiveTotal` don't time out. Unless the sending and receiving takes so long, that we run into a timeout here as well. But is that even possible when there is still an "active" stream meaning that funds are still transferred?